pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -0,0 +1,119 @@
+# coding=utf-8
+"""Tests upload of units via client."""
+from __future__ import unicode_literals
+
+import inspect
+import os
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import DRPM_URL, DRPM
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Execute ``pulp-admin login`` on the target Pulp system."""
+    set_up_module()
+    utils.pulp_admin_login(config.get_config())
+
+
+class UploadBaseTestCase(unittest2.TestCase):
+    """An abstract base class for test cases that uploads units to repository."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a repository."""
+        if inspect.getmro(cls)[0] == UploadBaseTestCase:
+            raise unittest2.SkipTest('Abstract base class.')
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo create --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the repository created by :meth:`setUpClass`."""
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo delete --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+        cli.Client(cls.cfg).run('pulp-admin orphan remove --all'.split())
+
+
+class UploadDrpmTestCase(UploadBaseTestCase):
+    """Test whether one can upload DRPMs."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Download temporary drpm."""
+        super(UploadDrpmTestCase, cls).setUpClass()
+
+        completed_process = cli.Client(cls.cfg).run('mktemp --directory'.split())
+        cls.temp_folder = completed_process.stdout.strip()
+        cls.temp_drpm = os.path.join(cls.temp_folder, DRPM)
+
+        cli.Client(cls.cfg).run(
+            'curl -o {} {}'
+            .format(cls.temp_drpm, DRPM_URL).split()
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the drpm created by :meth:`setUpClass`."""
+        cli.Client(cls.cfg).run('rm -rf {}'.format(cls.temp_folder).split())
```

Comment:
This doesn't call super. See what I mean about parent classes adding complexity? :wink: